examples/autoignition-mpi.py

```diff
@@ -240,71 +309,140 @@ def main(ctx_factory=cl.create_some_context, casename="autoignition", use_leap=F
             }
             from mirgecom.restart import write_restart_file
-            write_restart_file(actx, rst_data, rst_filename, comm)
-
-        # awful - computes potentially expensive viz quantities
-        #         regardless of whether it is time to viz
-        reaction_rates = eos.get_production_rates(state)
-        viz_fields = [("reaction_rates", reaction_rates)]
-        return sim_checkpoint(discr, visualizer, eos, cv=state,
-                              vizname=casename, step=step,
-                              t=t, dt=dt, nstatus=nstatus, nviz=nviz,
-                              constant_cfl=constant_cfl, comm=comm,
-                              viz_fields=viz_fields)
-
-    try:
-        (current_step, current_t, current_state) = \
-            advance_state(rhs=my_rhs, timestepper=timestepper,
-                checkpoint=my_checkpoint,
-                get_timestep=get_timestep, state=current_state,
-                t=current_t, t_final=t_final)
-    except ExactSolutionMismatch as ex:
-        error_state = True
-        current_step = ex.step
-        current_t = ex.t
-        current_state = ex.state
-
-    if not check_step(current_step, nviz):  # If final step not an output step
-        if rank == 0:
-            logger.info("Checkpointing final state ...")
-        my_checkpoint(current_step, t=current_t,
-                      dt=(current_t - checkpoint_t),
-                      state=current_state)
-
-    if current_t - t_final < 0:
-        error_state = True
-
-    if error_state:
-        raise ValueError("Simulation did not complete successfully.")
+            write_restart_file(actx, rst_data, rst_fname, comm)
+
+    def my_health_check(dv):
+        health_error = False
+        from mirgecom.simutil import check_naninf_local, check_range_local
+        if check_naninf_local(discr, "vol", dv.pressure) \
+           or check_range_local(discr, "vol", dv.pressure, 1e5, 2.4e5):
+            health_error = True
+            logger.info(f"{rank=}: Invalid pressure data found.")
+
+        if check_range_local(discr, "vol", dv.temperature, 1.4e3, 3.3e3):
+            health_error = True
+            logger.info(f"{rank=}: Invalid temperature data found.")
+
+        return health_error
+
+    def my_pre_step(step, t, dt, state):
+        try:
+            dv = None
+
+            if logmgr:
+                logmgr.tick_before()
+
+            from mirgecom.simutil import check_step
+            do_viz = check_step(step=step, interval=nviz)
+            do_restart = check_step(step=step, interval=nrestart)
+            do_health = check_step(step=step, interval=nhealth)
+
+            if do_health:
+                dv = eos.dependent_vars(state)
+                health_errors = my_health_check(dv)
+                if comm is not None:
+                    health_errors = comm.allreduce(health_errors, op=MPI.LOR)
+                if step == 5:   # quick test
+                    health_errors = True
```

Comment:
I assume this was left-over testing that can be removed?

Reply:
Yes - just wanting to demonstrate the health error while we have this PR cracked open.  I can certainly remove it right away if it is distracting.